controllers/chat/moveToAnon.js

```diff
@@ -48,11 +48,18 @@ const moveToAnon = async (req, res) => {
   const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
 
   let prevTargetUser;
-  if (oldChannelDetail[0].data.better_channel_member) {
+  if (
+    oldChannelDetail &&
+    oldChannelDetail[0] &&
+    oldChannelDetail[0].data &&
+    oldChannelDetail[0].data?.better_channel_member
+  ) {
     prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
       (user) => user.user_id !== mySignedId
     )[0];
     targetUserId = prevTargetUser.user_id;
+  } else {
+    return res.status(404).json(responseError('Old channel id not found'));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new checks for the existence of `oldChannelDetail` and its properties are a good addition to prevent potential `TypeError` when accessing undefined properties. However, there is a potential issue with the assignment of `targetUserId`. If `prevTargetUser` is undefined (i.e., the filter operation returns an empty array), trying to access `prevTargetUser.user_id` will throw a `TypeError`. To avoid this, you should check if `prevTargetUser` is defined before assigning `targetUserId`.

```diff
     prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
       (user) => user.user_id !== mySignedId
     )[0];
- 60:      targetUserId = prevTargetUser.user_id;
  } else {
    return res.status(404).json(responseError('Old channel id not found'));
   }
+  if (prevTargetUser) {
+    targetUserId = prevTargetUser.user_id;
+  } else {
+    return res.status(404).json(responseError('No other users in the old channel'));
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->